data/scripts/bin/update-hosts.sh.template

```diff
@@ -3,16 +3,24 @@ set -e
 
 BASE_URL="${SERVICE_BASE_URL}api/assisted-install/v2"
 
-while [[ "${cluster_status}" != "ready" ]]
+required_master_nodes=$(grep -m 1 required_master_nodes '/usr/local/bin/start-cluster-installation.sh' | awk -F"=" '{ print $2 }')
+required_worker_nodes=$(grep -m 1 required_worker_nodes '/usr/local/bin/start-cluster-installation.sh' | awk -F"=" '{ print $2 }')
+total_required_nodes=$(( ${required_master_nodes}+${required_worker_nodes} ))
+
+declare -a host_ids
+declare -i num_of_hosts
+while [[ "${num_of_hosts}" != "${total_required_nodes}" ]]
```

Comment:
won't it wait forever? don't we want to limit this by time?

Reply:
we can't continue the flow till it's done. so we must wait here till we have all nodes. same as done in the installer: https://github.com/openshift/installer/blob/f7731922a0f17a8339a3e837f72898ac77643611/data/data/agent/files/usr/local/bin/start-cluster-installation.sh.template#L49-L50